src/feature/dirauth/process_descs.c

```diff
@@ -221,50 +218,31 @@ dirserv_load_fingerprint_file(void)
         add_status = FP_INVALID;
     }
 
-    /* Check fingerprint type by key length, and if it's RSA or ed25519. */
-    int is_ed25519 = (strlen(fingerprint) == BASE64_DIGEST256_LEN);
-    int is_rsa = (strlen(fingerprint) == HEX_DIGEST_LEN);
-    bool is_valid_fpr = true;
+    /* Check if fingerprint is RSA or ed25519 by verifying it. */
+    int ed25519_not_ok, rsa_not_ok;
 
-    /* We only accept ed25519 or RSA keys */
-    if (!is_ed25519 && !is_rsa) {
-      is_valid_fpr = false;
+    /* Attempt to add the RSA key. */
+    rsa_not_ok = add_rsa_fingerprint_to_dir(fingerprint, fingerprint_list_new,
+                                            add_status);
```

Comment:
This code is unsafe: a malicious relay could have an ed25519 base64 encoded public key, that also passes the hexdigest check.

Please check the length of the string before passing it to this function.

Reply:
b6465c0aec08febe5d40dd343838095cb96faf1d